java/ql/src/semmle/code/java/security/ResponseSplitting.qll

```diff
@@ -0,0 +1,11 @@
+import semmle.code.java.dataflow.DataFlow
+
+/**
+ * Header-splitting sinks. Expressions that end up in an HTTP header.
+ */
+abstract class HeaderSplittingSink extends DataFlow::ExprNode { }
+
+/**
+ * Sources that cannot be used to perform a header splitting attack.
+ */
+abstract class TrustedSource extends DataFlow::ExprNode { }
```

Comment:
`TrustedSource` isn't a good name, since we don't trust it - we simply know that it's safe wrt. header splitting vulnerabilities.  Maybe something like `SafeHeaderSplittingSource` instead.  Also, we should extend `Node` rather than `ExprNode` here, and for clarity we ought to add a characteristic predicate that restricts this to `RemoteFlowSource` (note that we can't put it in `extends` it since it's `abstract`).